tensorboard/program.py

```diff
@@ -347,36 +349,67 @@ class WerkzeugServer(serving.ThreadedWSGIServer, TensorBoardServer):
   def __init__(self, wsgi_app, flags):
     self._flags = flags
     host = flags.host
+
+    # base_port: what's the first port to which we should try to bind?
+    # should_scan: if that fails, shall we try additional ports?
+    (base_port, should_scan) = (
```

Comment:
A multi-valued, multi-line ternary is kind of a complicated beast to read.  Maybe consider just doing

```python
should_scan = flags.port is None
base_port = flags.port if flags.port is not None else core_plugin.DEFAULT_PORT
```